test/utils.py

```diff
@@ -0,0 +1,42 @@
+from shlex import split
+from subprocess import run
+
+
+def determine_scope(
+    fixture_name,
+    config
+) -> str:
+    """
+    Sets the scope of certain fixtures.
+
+    Parameters
+    ----------
+    fixture_name
+        The name of the fixture.
+    config
+        A pytest internal object.
+
+    Returns
+    -------
+    str
+        The scope name.
+    """
+    if config.option.numprocesses in ["auto", "logical"]:
```

Comment:
```suggestion
    # XXX: right now, this always returns "function", but in theory, 
    # it could be better to return "session" when not run in parallel
    if config.option.numprocesses in ["auto", "logical"]:
```